docs/pages/docs/routing/middleware.mdx

```diff
@@ -593,29 +593,36 @@ Note that if you use a [`localePrefix`](#locale-prefix) other than `always`, you
 
 ### Example: Integrating with Clerk
 
-[`@clerk/nextjs`](https://clerk.com/docs/references/nextjs/overview) provides a middleware that can be integrated with `next-intl` by using the [`beforeAuth` hook](https://clerk.com/docs/references/nextjs/auth-middleware#using-before-auth-to-execute-middleware-before-authentication). By doing this, the middleware from `next-intl` will run first, potentially redirect or rewrite incoming requests, followed by the middleware from `@clerk/next` acting on the response.
+[`@clerk/nextjs`](https://clerk.com/docs/references/nextjs/overview) provides a middleware that can be [combined](https://clerk.com/docs/references/nextjs/clerk-middleware#combine-middleware) with other middlewares like the one provided by `next-intl`. By combining them, the middleware from `@clerk/next` will first ensure protected routes are handled appropriately. Subsequently, the middleware from `next-intl` will run, potentially redirecting or rewriting incoming requests.
 
 ```tsx filename="middleware.ts"
-import {authMiddleware} from '@clerk/nextjs';
+import {
+  clerkMiddleware,
+  createRouteMatcher,
+  redirectToSignIn,
+} from '@clerk/nextjs/server';
 import createMiddleware from 'next-intl/middleware';
 
 const intlMiddleware = createMiddleware({
   locales: ['en', 'de'],
   defaultLocale: 'en'
 });
 
-export default authMiddleware({
-  beforeAuth(request) {
-    return intlMiddleware(request);
-  },
+const isProtectedRoute = createRouteMatcher([
+  'dashboard/(.*)',
+]);
+
+export default clerkMiddleware((auth, req) => {
+  if (isProtectedRoute(req)) auth().protect();
 
-  // Ensure that locale-specific sign in pages are public
-  publicRoutes: ['/:locale', '/:locale/sign-in']
+  return intlMiddleware(req);
 });
 
 export const config = {
-  // Match only internationalized pathnames
-  matcher: ['/', '/(de|en)/:path*']
+  /**
+   * Match all paths except of internal /_next/ routes, static files and api paths
+   */
+  matcher: ['/((?!.+\\.[\\w]+$|_next).*)', '/', '/(api)(.*)'],
```

Comment:
Is there a need to update the matcher from the previous value? Generally, [we recommend a matcher](https://next-intl-docs.vercel.app/docs/routing/middleware#matcher-config) based on available locales as this is safer than checking for requests not containing a dot for example.

```suggestion
  // Match only internationalized pathnames
  matcher: ['/', '/(de|en)/:path*']
```

Reply:
The idea is that the clerk middleware runs on all page routes ensuring that auth checks are performed. Alternatively one could manually ensure that the matcher covers all protected routes, something like this:

```tsx filename="middleware.ts"
const isProtectedRoute = createRouteMatcher([
  '/(de|en)/dashboard(.*)',
]);

// ...

export const config = {
  // Match only internationalized pathnames
  matcher: ['/', '/(de|en)/:path*'],
};
```

Maybe the above would be good for the current example, while more sophisticated use cases should be handled individually. WDYT?

Reply:
Hmm, that's a good point.

Previously we had:

```tsx
  publicRoutes: ['/:locale', '/:locale/sign-in']
```

The updated docs now include:

```tsx
const isProtectedRoute = createRouteMatcher([
  'dashboard/(.*)',
]);
```

At least in the default config of `next-intl`, there should be a locale prefix for every route. That's also what the default matcher I've mentioned above assumes.

Does Clerk still support a pattern like `/:locale`? Or would this be `/[\w-]+/dashboard(.*)` with the new config to assume a locale prefix? If we don't need to hardcode locales that's a good thing, the Next.js `matcher` unfortunately requires a static config.

Do you have an app where you've upgraded to the latest Clerk version and is this working for you?

Reply:
Yes Clerk still does support the mentioned path pattern!

I upgraded a production app to the latest Clerk version without issues so far, unfortunately it is closed source. The biggest change IMHO ist that now by default, `clerkMiddleware` treats all pages as public.

I just created a test repo to play around, you can find it [here](https://github.com/gcascio/next-intl-clerk). There I used the `/:locale` pattern in the clerk middleware which looks like this:

```ts
import createMiddleware from 'next-intl/middleware';
import {
  clerkMiddleware,
  createRouteMatcher,
} from '@clerk/nextjs/server';

const intlMiddleware = createMiddleware({
  locales: ['en', 'de'],
  defaultLocale: 'en'
});

const isProtectedRoute = createRouteMatcher([
  '/:locale/dashboard(.*)',
]);

export default clerkMiddleware((auth, req) => {
  if (isProtectedRoute(req)) auth().protect();

  return intlMiddleware(req);
});

export const config = {
  // Match only internationalized pathnames
  matcher: ['/', '/(de|en)/:path*'],
};
```

I think this is as close as it gets to the previous example, if I am not missing something.

Reply:
Ah yes, that looks perfect from my perspective! Would you mind updating the PR again with that snippet? Happy to merge then!

Reply:
Great, I updated the PR!